File: crates/dewasm-backend-python/units/table/copy.py
# requires: rt/trap, table/check_range, table/slice
def copy(self, dst, other, src, length):
    if src + length > other.size():
        Rt.trap("out of bounds table access")
    self.check_range(dst, length)
    if length == 0:
        return
    self._slots[dst:dst + length] = other.slice(src, length)

File: crates/dewasm-backend-perl/units/table/copy.pl
# requires: rt/trap, table/check_range, table/slice
sub copy {
    my ($self, $dst, $other, $src, $len) = @_;
    Rt::trap('out of bounds table access') if $src + $len > $other->size();
    $self->check_range($dst, $len);
    return if $len == 0;
    my @tmp = $other->slice($src, $len);
    @{$self->{slots}}[$dst .. $dst + $len - 1] = @tmp;
}
